examples/apps/simple_wallet/workload/simple_wallet_logic.cpp

```diff
@@ -0,0 +1,154 @@
+        Log(TCF_LOG_INFO, "net Wallet Balance after deposit: %s",
+            netWalletBalance.c_str());
+        walletBalance = UpdateWalletBalance(accountName, netWalletBalance,
+            encryptionKey);
+        Log(TCF_LOG_INFO, "Updated Wallet Balance: %s",
+            walletBalance.c_str());
+    }
+    return accountName + "::" + walletBalance;
+}
+
+std::string TransferMoney(std::string debitorAccount, std::string creditorAccount,
+    std::string amount, std::string encryptionKey) {
+    std::string debitorBalance = ReadWalletBalance(debitorAccount,
+        encryptionKey);
+
+    if (debitorBalance.empty()) {
+        Log(TCF_LOG_ERROR, "No debitor account %s details found in the ledger: ",
+            debitorAccount.c_str());
+        return "Debitor wallet not found";
+    }
+
+    if (std::stoi(debitorBalance) < std::stoi(amount)) {
+        Log(TCF_LOG_ERROR, "No sufficient balance in debitor's account: %s",
+            debitorAccount.c_str());
+            return "No sufficient balance in debitor's account";
+    }
+    std::string creditorBalance = ReadWalletBalance(creditorAccount,
+        encryptionKey);
+    if (creditorBalance.empty()) {
+        Log(TCF_LOG_ERROR, "No creditor wallet details found for %s the ledger: ",
+            creditorAccount.c_str());
+        return "Creditor wallet not found";
+    }
+
+    std::string netDebitorBalance = std::to_string(std::stoi(debitorBalance) -
+        std::stoi(amount));
+    std::string netCreditorBalance = std::to_string(std::stoi(creditorBalance) +
+        std::stoi(amount));
+    std::string updatedDebitorBalance = UpdateWalletBalance(debitorAccount,
+        netDebitorBalance, encryptionKey);
+    std::string updatedCreditorBalance = UpdateWalletBalance(creditorAccount,
+        netCreditorBalance, encryptionKey);
+    return debitorAccount + "::" + updatedDebitorBalance + "," +
+        creditorAccount + "::" + updatedCreditorBalance;
+}
+
+std::string WithdrawMoney(std::string accountName, std::string amount,
+    std::string encryptionKey) {
+    std::string walletBalance = ReadWalletBalance(accountName, encryptionKey);
+
+    if (walletBalance.empty()) {
+        Log(TCF_LOG_ERROR, "No account %s details found in the ledger: ",
+            accountName.c_str());
+        return "No wallet detials found";
+    }
+
+    if (std::stoi(walletBalance) < std::stoi(amount)) {
+        Log(TCF_LOG_ERROR, "No sufficient balance to withdraw fund");
+        return "No sufficient balance to withdraw. Available balance: " +
+            walletBalance;
+    } else {
+        std::string netWalletBalance = std::to_string(std::stoi(walletBalance) -
+            std::stoi(amount));
+        walletBalance = UpdateWalletBalance(accountName, netWalletBalance, encryptionKey);
+        Log(TCF_LOG_DEBUG, "Net balance after withdrawal: %s", netWalletBalance);
+    }
+
+    return accountName + "::" + walletBalance;
+}
+
+std::string ProcessWalletRequest(std::string strIn) {
```

Comment:
Can you add a check for a bad format (that is, not a positive number) from the user. For example, entering an amount of `abc `instead of a number, such as `50`. Or is this done elsewhere?

Reply:
Ok, will add check.